src/three/plugins/images/ImageOverlayPlugin.js

```diff
@@ -1612,30 +1617,72 @@ export class CesiumIonOverlay extends ImageOverlay {
 		super( options );
 
 		const { apiToken, autoRefreshToken, assetId } = options;
+		this.options = options;
 		this.assetId = assetId;
 		this.auth = new CesiumIonAuth( { apiToken, autoRefreshToken } );
-		this.imageSource = new TMSImageSource( options );
 
 		this.auth.authURL = `https://api.cesium.com/v1/assets/${ assetId }/endpoint`;
-		this.imageSource.fetchData = ( ...args ) => this.fetch( ...args );
 		this._attributions = [];
 
+		this.externalType = null;
+
 	}
 
 	init() {
 
 		this._whenReady = this
 			.auth
 			.refreshToken()
-			.then( json => {
+			.then( async ( json ) => {
 
 				this._attributions = json.attributions.map( att => ( {
 					value: att.html,
 					type: 'html',
 					collapsible: att.collapsible,
 				} ) );
 
-				this.imageSource.url = json.url;
+				if ( json.type !== 'IMAGERY' ) throw new Error( 'Only IMAGERY is supported as overlay type' );
+
+				this.externalType = json.externalType;
+				switch ( json.externalType ) {
+
+					case 'GOOGLE_2D_MAPS':
+						this.imageSource = new XYZImageSource( {
+							...this.options,
+							url: `${json.options.url}/v1/2dtiles/{z}/{x}/{y}?session=${json.options.session}&key=${json.options.key}`,
+							tileDimension: json.options.tileWidth,
+						} );
```

Comment:
Is there a "known" number of tile levels for Google Map Tiles we can set here?

Reply:
This information is not specified within `options`. But Google says it goes up to 22 (https://developers.google.com/maps/documentation/tile/2d-tiles-overview?hl=en). So, it would be hard-coded here.